src/main/java/com/google/firebase/remoteconfig/RemoteConfigParameter.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.remoteconfig;
+
+import com.google.firebase.remoteconfig.internal.TemplateResponse.ParameterResponse;
+import com.google.firebase.remoteconfig.internal.TemplateResponse.ParameterValueResponse;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents a Remote Config parameter that can be included in a {@link RemoteConfigTemplate}.
+ * At minimum, a default value or a conditional value must be present for the
+ * parameter to have any effect.
+ */
+public final class RemoteConfigParameter {
+
+  private RemoteConfigParameterValue defaultValue;
+  private String description;
+  private Map<String, RemoteConfigParameterValue> conditionalValues;
+
+  /**
+   * Creates a new {@link RemoteConfigParameter}.
+   */
+  public RemoteConfigParameter() {
+    conditionalValues = new HashMap<>();
+    defaultValue = RemoteConfigParameterValue.of("");
+  }
+
+  /**
+   * Gets the default value of the parameter.
+   *
+   * @return A {@link RemoteConfigParameterValue} instance.
+   */
+  public RemoteConfigParameterValue getDefaultValue() {
+    return defaultValue;
+  }
+
+  /**
+   * Gets the description of the parameter.
+   *
+   * @return The {@link String} description of the parameter.
+   */
+  public String getDescription() {
+    return description;
+  }
+
+  /**
+   * Gets the conditional values of the parameter.
+   * The condition name of the highest priority (the one listed first in the
+   * {@link RemoteConfigTemplate}'s conditions list) determines the value of this parameter.
+   *
+   * @return A map of conditional values.
+   */
+  public Map<String, RemoteConfigParameterValue> getConditionalValues() {
+    return conditionalValues;
+  }
+
+  /**
+   * Sets the default value of the parameter.
+   * This is the value to set the parameter to, when none of the named conditions
```

Comment:
Does the developer actually set the value?  Or does our backend logic?

Reply:
When getting an existing template the backend will set the values for existing Parameters. If the developer wants to create and add new parameters then they can set a default value to set the parameter to, when none of the named conditions evaluate to true.